File: LC1232.java
class Solution {
    public boolean checkStraightLine(int[][] coordinates) {
        if(coordinates.length<=2)
            return true;
        boolean flag=true;
        for(int i=0;i<coordinates.length-1;i++)
        {
            if(coordinates[i][0]!=coordinates[i+1][0])
            {
                flag=false;
                break;
            }
        }
        if(flag)
            return true;
        flag=true;
        for(int i=0;i<coordinates.length-1;i++)
        {
            if(coordinates[i][1]!=coordinates[i+1][1])
            {
                flag=false;
                break;
            }
        }
        if(flag)
            return true;
        flag=true;
        for(int i=0;i<coordinates.length-2;i++)
        {
            int x1=coordinates[i+1][0]-coordinates[i][0];
            int y1=coordinates[i+1][1]-coordinates[i][1];
            int x2=coordinates[i+2][0]-coordinates[i+1][0];
            int y2=coordinates[i+2][1]-coordinates[i+1][1];
            if(x1*y2!=x2*y1)
            {
                flag=false;
                break;
            }
        }
        return flag;
    }
}

File: LC1232.go
package main

func checkStraightLine(coordinates [][]int) bool {
	if len(coordinates) <= 2 {
		return true
	}
	x1 := coordinates[0][0]
	x2 := coordinates[1][0]
	y1 := coordinates[0][1]
	y2 := coordinates[1][1]

	a := x1 - x2
	b := -y1 * (x1 - x2)
	c := y1 - y2
	d := -x1 * (y1 - y2)

	for i := 2; i < len(coordinates); i++ {
		x := coordinates[i][0]
		y := coordinates[i][1]
		if a*y+b != c*x+d {
			return false
		}
	}
	return true
}
